src/main/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfiguration.java

```diff
@@ -163,16 +164,25 @@ public record Synapse(
       int connectRetryInterval,
       int connectRetryCount) {}
 
-  public record Threading(int numTableThreads, int maxQueueSize) {}
+  public record Threading(int numTableThreads, int maxQueueSize, boolean blockWhenSaturated) {}
 
-  @Bean("azureTableThreadpool")
+  /**
+   * The name of a {@link AsyncTaskExecutor} Spring Bean which executes tasks related to Azure
+   * Storage Tables.
+   */
+  public static final String TABLE_THREADPOOL_NAME = "azureTableThreadpool";
```

Comment:
This is fine, but I wonder if having a subclass (or wrapper class) would be easier to use than a qualifier. For example, a subclass could be written as:
```
@Component
public class AzureTableThreadPool extends ThreadPoolTaskExecutor {

  public AzureTableThreadPool(AzureResourceConfiguration azureResourceConfiguration) {
    var threading = azureResourceConfiguration.threading();

    setCorePoolSize(threading.numTableThreads());
    setMaxPoolSize(threading.numTableThreads());
    setKeepAliveSeconds(0);
    setQueueCapacity(threading.maxQueueSize());
    setThreadNamePrefix("az-table-thread-");
    if (threading.blockWhenSaturated()) {
      executor.setRejectedExecutionHandler(new BlockingRejectedExecutionHandler());
    }
    initialize();
  }
}
```

Reply:
I liked this suggestion and started to pursue it, but found that changes to tests it required make it more than I want to take on within this PR.